psi4/src/psi4/libsapt_solver/amplitudes.cc

```diff
@@ -620,7 +620,7 @@ void SAPT2::ijkl_to_ikjl(double *tARAR, int ilength, int jlength, int klength,
 
   for(int i=0; i<ilength; i++) {
     for(int l=0; l<llength; l++) {
-      long int ijkl = i*jlength*klength*llength + l;
+      long int ijkl = (long int)i*jlength*klength*llength + l;
```

Comment:
I might remember wrong, but don't you need to also cast jlength, klength and llength, since the order of evaluation isn't guaranteed...? What this is doing is typecasting i to long int, but jlength*klength*llength is still int and might overflow.